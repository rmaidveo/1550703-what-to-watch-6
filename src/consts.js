const StarList = {
  MIN: 0,
  MAX: 3
};
const GeneresCount = {
  MIN: 0,
  MAX: 9,
};
const RATING_STARS = 10;
const FILMS_COUNT = 8;
const TabsTypes = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
const RouteType = {
  INDEX: `/`,
  LOGIN: `/login`,
  LOG_OUT: `/logout`,
  USER_LIST: `/mylist`,
  PLAYER: `/player/:id?`,
  REVIEW: `/films/:id?/review`,
  FILM_PAGE: `/films/:id?`
};
const GENERE_ALL = `All genres`;
const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
const TextArea = {
  MIN: 40,
  MAX: 400
};

export {
  RATING_STARS,
  StarList,
  TabsTypes,
  RouteType,
  GENERE_ALL,
  GeneresCount,
  FILMS_COUNT,
  AuthorizationStatus,
  TextArea
};
