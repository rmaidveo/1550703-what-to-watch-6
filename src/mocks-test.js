export const films = [
  {
    id: 1,
    name: `Bohemian Rhapsody`,
    imageSrc: `img/bohemian-rhapsody.jpg`,
    previewImageSrc: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 99,
    genre: `Drama`,
    released: 2014,
    isFavorite: true
  },
  {
    id: 2,
    name: `Macbeth`,
    imageSrc: `img/macbeth.jpg`,
    previewImageSrc: `img/macbeth.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    released: 2014,
    isFavorite: true
  }
];
export const film = {
  id: 1,
  name: `Bohemian Rhapsody`,
  imageSrc: `img/bohemian-rhapsody.jpg`,
  previewImageSrc: `img/bohemian-rhapsody.jpg`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`],
  runTime: 99,
  genre: `Drama`,
  released: 2014,
  isFavorite: true,
};

export const reviews = [
  {
    "id": 1,
    "user": {
      "id": 10,
      "name": `Max`
    },
    "rating": 3.9,
    "comment": `Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.`,
    "date": `2021-02-18T08:04:28.658Z`
  },
  {
    "id": 2,
    "user": {
      "id": 19,
      "name": `Christina`
    },
    "rating": 9.7,
    "comment": `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    "date": `2021-02-17T08:04:28.658Z`
  }
];
