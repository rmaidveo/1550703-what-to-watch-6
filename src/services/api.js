import axios from 'axios';
import {toast} from '../utils/toast/toast';

const BACKEND_URL = `https://6.react.pages.academy/wtw`;
const REQUEST_TIMEOUT = 5000;
const ERROR_SERVER_CONNECTION = `Server connection error, try again later`;

const HttpCode = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
  NOT_FOUND: 404,
  UNKNOWN: 0
};

export const createAPI = (onUnauthorized, onRedirect) => {
  const api = axios.create({
    baseURL: BACKEND_URL,
    timeout: REQUEST_TIMEOUT,
    withCredentials: true,
  });

  const onSuccess = (response) => response;

  const onFail = (err) => {
    const {response} = err;

    if (err.request.status === HttpCode.UNKNOWN) {
      toast(`Error: ${err.message}`);
      throw err;
    }

    if (response.status === HttpCode.UNAUTHORIZED) {
      onUnauthorized();
      throw err;
    }

    if (response.status === HttpCode.NOT_FOUND) {
      onRedirect();
      throw err;
    }

    if (response.status === HttpCode.BAD_REQUEST) {
      toast(ERROR_SERVER_CONNECTION);
      throw err;
    }

    throw err;
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
