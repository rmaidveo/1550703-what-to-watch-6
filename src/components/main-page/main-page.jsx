import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {mainPagePropTypes} from '../../prop-types';
import Header from '../header/header';
import Footer from '../footer/footer.jsx';
import Catalog from '../catalog/catalog';
import {getPromo} from '../../store/films/selectors';
import {AuthorizationStatus} from '../../consts';
import {getAuthorizationStatus} from '../../store/user/selectors';
import AddInList from '../add-in-list/add-in-list';

const MainPage = (props) => {
  const {promo, authorizationStatus} = props;

  return (
    <>
      <section className="movie-card">
        <div className="movie-card__bg">
          <img src={promo.backgroundImage} alt={promo.name}/>
        </div>
        <Header/>
        <div className="movie-card__wrap">
          <div className="movie-card__info">
            <div className="movie-card__poster">
              <img src={promo.imageSrc} alt={promo.name + `poster`} width="218" height="327" />
            </div>

            <div className="movie-card__desc">
              <h2 className="movie-card__title">{promo.name}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{promo.genre}</span>
                <span className="movie-card__year">{promo.released}</span>
              </p>

              <div className="movie-card__buttons">
                <Link to={`/player/${promo.id}`} className="btn btn--play movie-card__button" type="button">
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use xlinkHref="#play-s"> </use>
                  </svg>
                  <span>Play</span>
                </Link>
                {authorizationStatus === AuthorizationStatus.AUTH ?
                  <AddInList id={promo.id} isFavorite={promo.isFavorite}/>
                  : ``}
              </div>
            </div>
          </div>
        </div>
      </section>
      <div className="page-content">
        <Catalog />
        <Footer/>
      </div>
    </>
  );
};

MainPage.propTypes = mainPagePropTypes;

const mapStateToProps = (state) => ({
  promo: getPromo(state),
  authorizationStatus: getAuthorizationStatus(state),
});

export {MainPage};
export default connect(mapStateToProps, null)(MainPage);
