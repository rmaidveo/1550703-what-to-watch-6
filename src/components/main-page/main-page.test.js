import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import {createMemoryHistory} from 'history';
import {MainPage} from './main-page';
import {film, films} from '../../mocks-test';
import {AuthorizationStatus, FILMS_COUNT, GENERE_ALL} from '../../consts';
import thunk from 'redux-thunk';
import * as redux from "react-redux";
import {createAPI} from '../../services/api';

const api = createAPI(() => {});
const middlleware = thunk.withExtraArgument(api);
const mockStore = configureStore([middlleware]);

it(`MainScreen renders correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    FILMS: {
      promo: film,
      films,
      favoriteFilmsList: films,
      isApplycationReady: true,
      visibleFilms: FILMS_COUNT,
      genre: GENERE_ALL
    },
    USER: {
      authorizationStatus: AuthorizationStatus.AUTH,
    },
  });

  const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <MainPage
            isApplicationReady={true}
            authorizationStatus={AuthorizationStatus.AUTH}
            promo={film}
          />
        </Router>
      </redux.Provider>
  );
  expect(container).toMatchSnapshot();
});
