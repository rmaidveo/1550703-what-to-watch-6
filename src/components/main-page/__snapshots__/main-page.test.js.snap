// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen renders correctly 1`] = `
<div>
  <section
    class="movie-card"
  >
    <div
      class="movie-card__bg"
    >
      <img
        alt="Bohemian Rhapsody"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      class="visually-hidden"
    >
      WTW
    </h1>
    <header
      class="page-header movie-card__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link undefined"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="user-block"
      >
        <div
          class="user-block__avatar"
        >
          <img
            alt="User avatar"
            data-testid="my-list"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
        <div
          class="user-block"
        >
          <span
            class="movie-card__year"
          >
             
             
          </span>
          <a
            class="user-block__link"
            href="#"
          >
            Log out
          </a>
        </div>
      </div>
    </header>
    <div
      class="movie-card__wrap"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster"
        >
          <img
            alt="Bohemian Rhapsodyposter"
            height="327"
            src="img/bohemian-rhapsody.jpg"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          >
            Bohemian Rhapsody
          </h2>
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Drama
            </span>
            <span
              class="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            class="movie-card__buttons"
          >
            <a
              class="btn btn--play movie-card__button"
              href="/player/1"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                >
                   
                </use>
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlink:href="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog"
    >
      <h2
        class="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        class="catalog__genres-list"
      >
        <li
          class="catalog__genres-item catalog__genres-item--active"
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          class="catalog__genres-item"
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            Comedy
          </a>
        </li>
        <li
          class="catalog__genres-item"
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            Drama
          </a>
        </li>
      </ul>
      <div
        class="catalog__movies-list"
        data-testid="film-card-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/1"
          >
            <div
              class="small-movie-card__image"
            >
              <img
                alt="Bohemian Rhapsody"
                height="175"
                src="img/bohemian-rhapsody.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            class="small-movie-card__title"
            data-testid="film-card-name"
          >
            <a
              class="small-movie-card__link"
              href="/films/1"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/2"
          >
            <div
              class="small-movie-card__image"
            >
              <img
                alt="Macbeth"
                height="175"
                src="img/macbeth.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            class="small-movie-card__title"
            data-testid="film-card-name"
          >
            <a
              class="small-movie-card__link"
              href="/films/2"
            >
              Macbeth
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          data-testid="logo__link--light"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
